packages/gatsby/src/utils/api-runner-node.js

```diff
@@ -119,6 +128,8 @@ const runAPI = (plugin, api, args) => {
         createNodeId: namespacedCreateNodeId,
         createContentDigest,
         tracing,
+        withPrefixPath: withPrefixPath(pathPrefix),
+        withAssetPath: withAssetPath(assetPrefix),
```

Comment:
The idea here is that any assets (images, css, js, etc.) run outside of webpack will need to use this helper, which will be automatically bound to the assetPrefix, if it exists.

Then, in a lifecycle hook, e.g. `onPostBuild`, we'll make available an `assets` property which can be used to copy/interact with these assets, as needed.

For instance, if our cdn is hosted at `https://cdn.gatsbyjs.org`, we should set that up as our asset prefix. Plugins (and webpack) would use this _for free_, and then we would copy/move our assets to this however we prefer, e.g. a CLI for s3.